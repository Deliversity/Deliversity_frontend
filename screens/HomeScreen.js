import React from 'react';
import {View, Text, Button, StyleSheet, StatusBar, Image} from 'react-native';
import {useTheme} from '@react-navigation/native';

const HomeScreen = ({navigation}) => {
  const {colors} = useTheme();

  const theme = useTheme();

  return (
    <View style={styles.container}>
      <StatusBar barStyle={theme.dark ? 'light-content' : 'dark-content'} />
      <Image
        source={require('../assets/logo.jpg')}
        style={{width: 300, height: 200, marginRight: 10}}
      />
    </View>
  );
};

export default HomeScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  }
});
