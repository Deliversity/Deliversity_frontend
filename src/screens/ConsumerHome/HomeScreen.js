import React, {Component} from 'react';
import {
  Image,
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Modal,
  Dimensions,
  Platform,
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import Store from './StoreScreen';
import axios from '../../axiosConfig';
import ChangeButton from '../../components/ChangeButton';
import Geolocation from '@react-native-community/geolocation';
import {setUserStorage} from '../../store/actions/action';
import HomeInfo from '../../../assets/homeDetail.png';
type Props = {};
const minusHeight = Platform.OS === 'ios' ? 123 : 14;
const {width, height} = Dimensions.get('window');
export default class App extends Component<Props> {
  state = {
    modalVisible: true,
  };

  onClickCategory = async (data) => {
    this.props.navigation.navigate('Store', {
      category: data,
    });
  };

  geo = async () => {
    Geolocation.getCurrentPosition(async (position) => {
      await axios.post('/api/v1/myinfo/currentLocation', position);
    });
  };

  startPostPosition = async () => {
    console.log('geo');
    const timerId = setInterval(this.geo, 1000 * 60 * 3);
    await setUserStorage('timerId', timerId.toString());
  };

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.submitBtn}>
          <TouchableOpacity
            onPress={() => this.setState({modalVisible: false})}
            style={{marginRight: 10, marginTop: 10}}>
            <Text
              style={{
                color: 'white',
                textAlign: 'center',
                marginBottom: 6,
                fontSize: 15,
              }}>
              ❓
            </Text>
            <Text style={{color: 'white', fontWeight: 'bold', fontSize: 11}}>
              도움말
            </Text>
          </TouchableOpacity>
          <ChangeButton onPress={this.startPostPosition} />
        </View>
        <View style={styles.header}>
          <Image
            source={require('../../../assets/Deliver2.png')}
            
          />
        </View>
        <View style={styles.footer}>
          <Text style={{textAlign: 'center', fontSize: 15}}>
            카테고리를 선택하여 주문을 시작하세요!
          </Text>
          <View
            style={{
              alignSelf: 'center',
              borderTopWidth: 5,
              borderTopColor: '#8fbc8f',
              width: '85%',
              borderTopRightRadius: 15,
              borderTopLeftRadius: 15,
            }}
          />
          <View style={styles.content}>
            <View style={styles.elem}>
              <View style={styles.userInfo}>
                <TouchableOpacity
                  onPress={() => {
                    this.onClickCategory('편의점');
                  }}>
                  <View style={styles.profile}>
                    <Icon
                      name="local-convenience-store"
                      color={'#000000'}
                      size={50}
                    />
                    <Text style={styles.name}>편의점</Text>
                  </View>
                </TouchableOpacity>
              </View>
              <View style={styles.userInfo}>
                <TouchableOpacity
                  onPress={() => {
                    this.onClickCategory('커피전문점');
                  }}>
                  <View style={styles.profile}>
                    <Icon name="local-cafe" color={'#000000'} size={50} />
                    <Text style={styles.name}>카페</Text>
                  </View>
                </TouchableOpacity>
              </View>
              <View style={styles.userInfo}>
                <TouchableOpacity
                  onPress={() => {
                    this.onClickCategory('음식점');
                  }}>
                  <View style={styles.profile}>
                    <Icon name="local-restaurant" color={'#000000'} size={50} />
                    <Text style={styles.name}>음식점</Text>
                  </View>
                </TouchableOpacity>
              </View>
            </View>
            <View style={styles.elem}>
              <View style={styles.userInfo}>
                <TouchableOpacity
                  onPress={() => {
                    this.onClickCategory('세탁소');
                  }}>
                  <View style={styles.profile}>
                    <Icon
                      name="local-laundry-service"
                      color={'#000000'}
                      size={50}
                    />
                    <Text style={styles.name}>세탁소</Text>
                  </View>
                </TouchableOpacity>
              </View>
              <View style={styles.userInfo}>
                <TouchableOpacity
                  onPress={() => {
                    this.onClickCategory('문구');
                  }}>
                  <View style={styles.profile}>
                    <Icon name="local-printshop" color={'#000000'} size={50} />
                    <Text style={styles.name}>문구점</Text>
                  </View>
                </TouchableOpacity>
              </View>
              <View style={styles.userInfo}>
                <TouchableOpacity
                  onPress={() => {
                    this.onClickCategory('기타');
                  }}>
                  <View style={styles.profile}>
                    <Icon name="add" color={'#ffff'} size={50} />
                    <Text style={styles.name}>기타</Text>
                  </View>
                </TouchableOpacity>
              </View>
            </View>
          </View>
        </View>
        <Modal
          animationType="slide"
          transparent={true}
          visible={this.state.modalVisible === false}>
          <View style={{opacity: 0.7, backgroundColor: 'white'}}>
            <TouchableOpacity
              onPress={() => this.setState({modalVisible: true})}
              style={{marginLeft: 10, marginTop: 10}}>
              <Icon name="clear" color={'#000000'} size={25} />
            </TouchableOpacity>
            <Image
              source={HomeInfo}
              style={{height: height - minusHeight, width: width}}
            />
          </View>
        </Modal>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#8fbc8f',
  },
  header: {
    flex: 2,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    paddingBottom: 30,
  },
  text_header: {
    color: '#fff',
    fontWeight: 'bold',
    fontSize: 30,
  },
  footer: {
    flex: 4,
    backgroundColor: '#fff',
    borderTopLeftRadius: 30,
    borderTopRightRadius: 30,
    paddingHorizontal: 20,
    paddingVertical: 30,
  },
  content: {
    flex: 1,
    justifyContent: 'center',
  },

  elem: {
    width: '100%',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    padding: 5,
  },
  userInfo: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  profile: {
    width: 100,
    height: 100,
    borderRadius: 50,
    alignItems: 'center',
    justifyContent: 'center',
    textAlign: 'center',
    backgroundColor: '#ffd700',
  },
  name: {
    fontWeight: 'bold',
  },
  submitBtn: {
    alignSelf: 'flex-end',
    flexDirection: 'row',
  },
});
