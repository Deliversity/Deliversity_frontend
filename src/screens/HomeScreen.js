import React, {Component} from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
type Props = {};
export default class App extends Component<Props> {
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.header}>
          <Image
            source={require('../../assets/logo.png')}
            style={{width: 200, height: 200}}
          />
          <Text style={styles.text_header}>Deliversity</Text>
        </View>
        <View style={styles.footer}>
          <View style={styles.content}>
            <View style={styles.elem}>
              <View style={styles.userInfo}>
                <View style={styles.profile}>
                  <Icon
                    name="local-convenience-store"
                    color={'#000000'}
                    size={50}
                  />
                  <Text style={styles.name}>편의점</Text>
                </View>
              </View>
              <View style={styles.userInfo}>
                <View style={styles.profile}>
                  <Icon name="local-pharmacy" color={'#000000'} size={50} />
                  <Text style={styles.name}>약국</Text>
                </View>
              </View>
              <View style={styles.userInfo}>
                <View style={styles.profile}>
                  <Icon
                    name="local-laundry-service"
                    color={'#000000'}
                    size={50}
                  />
                  <Text style={styles.name}>세탁물</Text>
                </View>
              </View>
            </View>
            <View style={styles.elem}>
              <View style={styles.userInfo}>
                <View style={styles.profile}>
                  <Icon name="local-restaurant" color={'#000000'} size={50} />
                  <Text style={styles.name}>음식</Text>
                </View>
              </View>
              <View style={styles.userInfo}>
                <View style={styles.profile}>
                  <Icon name="local-printshop" color={'#000000'} size={50} />
                  <Text style={styles.name}>문구점</Text>
                </View>
              </View>
              <View style={styles.userInfo}>
                <View style={styles.profile}>
                  <Icon name="add" color={'#ffff'} size={50} />
                  <Text style={styles.name}>기타</Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#8fbc8f',
  },
  header: {
    flex: 3,
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'center',
    paddingHorizontal: 20,
    paddingBottom: 5,
  },
  text_header: {
    color: '#fff',
    fontWeight: 'bold',
    fontSize: 30,
  },
  footer: {
    flex: 4,
    backgroundColor: '#fff',
    borderTopLeftRadius: 30,
    borderTopRightRadius: 30,
    paddingHorizontal: 20,
    paddingVertical: 30,
  },
  content: {
    flex: 1,
    justifyContent: 'center',
  },

  elem: {
    width: '100%',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    padding: 5,
  },
  userInfo: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  userComment: {
    padding: 8,
    backgroundColor: '#ffd700',
    borderRadius: 5,
  },
  profile: {
    width: 100,
    height: 100,
    borderRadius: 50,
    alignItems: 'center',
    justifyContent: 'center',
    textAlign: 'center',
    backgroundColor: '#ffd700',
  },
  name: {
    fontWeight: 'bold',
  },
});
