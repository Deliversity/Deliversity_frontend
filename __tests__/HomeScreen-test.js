import 'react-native';

import React, {ReactElement} from 'react';
import HomeScreen from '../src/screens/HomeScreen';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Provider} from 'react-redux';
const mockStore = configureMockStore();
let props;
let component: ReactElement;
configure({adapter: new Adapter()});
beforeAll(() => {
  jest.mock('@react-native-community/async-storage');
  // jest.mock('@react-native-firebase');
});

describe('[Temp] render', () => {
  let store;
  props = {}; // fill test props
  beforeEach(() => {
    const initialState = {
      authentication: {user: '사용자'},
    };
    store = mockStore(initialState);
    component = shallow(
      <Provider store={store}>
        <HomeScreen />
      </Provider>,
    );
  });
  it('renders without crashing', () => {
    const renderResult = renderer.create(component);
    const renderedJson = renderResult.toJSON();
    expect(renderedJson).toMatchSnapshot();
    expect(renderedJson).toBeTruthy();
  });
});
