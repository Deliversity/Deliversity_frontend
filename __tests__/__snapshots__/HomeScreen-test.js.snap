// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Temp] render renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#8fbc8f",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "flex-end",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#f4da6c",
                "fontSize": 35,
              },
              undefined,
              Object {
                "fontFamily": "MaterialIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "color": "#ffff",
              "fontSize": 11,
              "fontWeight": "bold",
            }
          }
        >
          사용자
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 30,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/logo_D.png",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    >
      Deliversity
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderTopLeftRadius": 30,
        "borderTopRightRadius": 30,
        "flex": 4,
        "paddingHorizontal": 20,
        "paddingVertical": 30,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 5,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffd700",
                  "borderRadius": 50,
                  "height": 100,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "width": 100,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontSize": 50,
                    },
                    undefined,
                    Object {
                      "fontFamily": "MaterialIcons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                편의점
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffd700",
                  "borderRadius": 50,
                  "height": 100,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "width": 100,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontSize": 50,
                    },
                    undefined,
                    Object {
                      "fontFamily": "MaterialIcons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                카페
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffd700",
                  "borderRadius": 50,
                  "height": 100,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "width": 100,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontSize": 50,
                    },
                    undefined,
                    Object {
                      "fontFamily": "MaterialIcons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                음식점
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 5,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffd700",
                  "borderRadius": 50,
                  "height": 100,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "width": 100,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontSize": 50,
                    },
                    undefined,
                    Object {
                      "fontFamily": "MaterialIcons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                세탁소
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffd700",
                  "borderRadius": 50,
                  "height": 100,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "width": 100,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontSize": 50,
                    },
                    undefined,
                    Object {
                      "fontFamily": "MaterialIcons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                문구점
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffd700",
                  "borderRadius": 50,
                  "height": 100,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "width": 100,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#ffff",
                      "fontSize": 50,
                    },
                    undefined,
                    Object {
                      "fontFamily": "MaterialIcons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                기타
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
